Makefile

```diff
@@ -1,26 +1,115 @@
-.PHONY: docs
+SHELL := /bin/bash
+# =============================================================================
+# Variables
+# =============================================================================
 
-docs-clean:
-	rm -rf docs/_build
+.DEFAULT_GOAL:=help
+.ONESHELL:
+USING_PDM		=	$(shell grep "tool.pdm" pyproject.toml && echo "yes")
+ENV_PREFIX		=	$(shell python3 -c "if __import__('pathlib').Path('.venv/bin/pip').exists(): print('.venv/bin/')")
+VENV_EXISTS		=	$(shell python3 -c "if __import__('pathlib').Path('.venv/bin/activate').exists(): print('yes')")
+PDM_OPTS 		?=
+PDM 			?= 	pdm $(PDM_OPTS)
 
-docs-serve:
-	sphinx-autobuild docs docs/_build/ -j auto --watch litestar,examples
+.EXPORT_ALL_VARIABLES:
 
-docs: docs-clean
-	sphinx-build -M html docs docs/_build/ -a -j auto -W --keep-going
 
-docs-test:
-	rm -rf test_docs/_build
-	sphinx-build -M html test_docs test_docs/_build/ -a
+.PHONY: help
+help: 		   										## Display this help text for Makefile
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 
-test-examples:
+.PHONY: upgrade
+upgrade:       										## Upgrade all dependencies to the latest stable versions
+	@echo "=> Updating all dependencies"
+	@if [ "$(USING_PDM)" ]; then $(PDM) update; fi
+	@echo "=> Dependencies Updated"
+	@$(ENV_PREFIX)pre-commit autoupdate
+	@echo "=> Updated Pre-commit"
+
+# =============================================================================
+# Developer Utils
+# =============================================================================
+.PHONY: install-pdm
+install-pdm: 										## Install latest version of PDM
```

Comment:
I think it's a good idea to add a checksum verification as recommended in the installation guide by PDM.